examples/homography_estimation.py

```diff
@@ -0,0 +1,463 @@
+import kornia
+from typing import List
+import glob
+import theseus as th
+import torch
+import torch.nn as nn
+from PIL import Image
+import cv2
+import numpy as np
+import os
+from torch.utils.data import DataLoader, Dataset
+import shutil
+from torch.utils.tensorboard import SummaryWriter
+
+from libs.easyaug import RandomGeoAug, GeoAugParam, RandomPhotoAug
+
+FONT = cv2.FONT_HERSHEY_DUPLEX
+FONT_SZ = 0.5
+FONT_PT = (5, 15)
+
+BACKWARD_MODE = {
+    "implicit": th.BackwardMode.IMPLICIT,
+    "full": th.BackwardMode.FULL,
+    "truncated": th.BackwardMode.TRUNCATED,
+}
+
+
+def prepare_data():
+    dataset_root = os.path.join(os.getcwd(), "data")
+    chunks = [
+        "revisitop1m.1",
+        "revisitop1m.2",
+        "revisitop1m.3",
+        "revisitop1m.4",
+        "revisitop1m.5",
+    ]
+    dataset_paths = []
+    for chunk in chunks:
+        dataset_path = os.path.join(dataset_root, chunk)
+        dataset_paths.append(dataset_path)
+        if not os.path.exists(dataset_path):
+            print("Downloading data")
+            url_root = "http://ptak.felk.cvut.cz/revisitop/revisitop1m/jpg/"
+            tar = "%s.tar.gz" % chunk
+            cmd = "wget %s/%s -O %s/%s" % (url_root, tar, dataset_root, tar)
+            print("Running command: ", cmd)
+            os.system(cmd)
+            os.makedirs(dataset_path)
+            cmd = "tar -xf %s/%s -C %s" % (dataset_root, tar, dataset_path)
+            print("Running command: ", cmd)
+            os.system(cmd)
+
+    return dataset_paths
+
+
+class HomographyDataset(Dataset):
```

Comment:
Can we add some brief comments to all functions describing what they do? (both in/out of this class). Nothing too elaborate, just a couple of sentences. @ddetone @joeaortiz

Reply:
Remember to also appropriately update the examples/readme file.

Reply:
I've added some comments now describing the functions.